core/src/main/java/org/axonframework/messaging/StreamableMessageSource.java

```diff
@@ -35,4 +38,49 @@
      */
     MessageStream<M> openStream(TrackingToken trackingToken);
 
+    /**
+     * Creates the token at the beginning of an event stream. The begging of an event stream in this context means the
+     * token of very first event in the stream.
+     *
+     * @return the token at the beginning of an event stream
+     */
+    default TrackingToken createTailToken() {
+        return null;
```

Comment:
Although I know this works fine as the start of the stream for the framework, I'd like to argue that `null` does not equal a '`TrackingToken` at the beginning of the event stream'.

Reply:
Thinking of it, we should orevent `null` being returned. If anyone calls one of the operationa on the returned value, they’ll face a NPE. 
I think all implementations should be capable of returning a non-`null` token that represents the beinning of a stream.

Reply:
Ok, everything is clear. I'll query for the beginning of stream instead of returning `null` here.